lib/cartopy/mpl/geoaxes.py

```diff
@@ -585,7 +585,18 @@ def set_extent(self, extents, crs=None):
 
         if projected is None:
             projected = self.projection.project_geometry(domain_in_crs, crs)
-        x1, y1, x2, y2 = projected.bounds
+        try:
+            # This might fail with an unhelpful error message ('need more
+            # than 0 values to unpack') if the specified extents fall outside
+            # the projection extents, so try and give a better error message.
+            x1, y1, x2, y2 = projected.bounds
```

Comment:
What is ``projected.bounds`` when the error message occurs? Is it just an empty list? If so, we can just test for it upfront. That way, if we get any real ValueErrors, we don't lose that detail.